.gitignore

```diff
@@ -1,5 +1,249 @@
-build.sh
+# Created by https://www.toptal.com/developers/gitignore/api/cmake,python,c++,linux
+# Edit at https://www.toptal.com/developers/gitignore?templates=cmake,python,c++,linux
+
+### C++ ###
+# Prerequisites
+*.d
+
+# Compiled Object files
+*.slo
+*.lo
+*.o
+*.obj
+
+# Precompiled Headers
+*.gch
+*.pch
+
+# Compiled Dynamic libraries
+*.so
+*.dylib
+*.dll
+
+# Fortran module files
+*.mod
```

Comment:
Is this a file we use in all our projects?
Be careful with this one, *.mod files are also files used in the Neuron simulator to describe channel kinetics, and should not be ignored.

Reply:
OK, i did not know that. we can strip it away as well. 
I also understand that CMake can redirect all artifacts to a single target dir and many of these are not needed.
The list above was auto-generated to ignore artifacts from cmake, python, c++, linux (some links at the top of the file) 

We typically start with these gitignore listings because we have many people building different tech on our repos but if you are well organized with CMake i totally get you miht not want this.

please do not hesitate to reject the PR if you think this change is unnecesary.